File: trunk/ObjectMapper/GUI/Components/Docking/DockingForm.vb
Imports WeifenLuo.WinFormsUI

Public Class DockingForm
    Inherits DockContent


    Public Sub New()
        Me.HideOnClose = True
        Me.DockableAreas = DockAreas.DockBottom Or DockAreas.DockLeft Or DockAreas.DockRight Or DockAreas.DockTop Or DockAreas.Document Or DockAreas.Float
    End Sub


    Private content As Panel
    Private oldParent As Control
    Public Sub SetContent(ByVal content As Panel, ByVal title As String)

        Me.HideOnClose = True

        Me.content = content
        Me.oldParent = content.Parent

        Me.Controls.Clear()
        content.Parent = Me
        content.Dock = DockStyle.Fill
        Me.Text = title
        content.Visible = True
    End Sub

End Class

File: trunk/Experimental/GenerationStudio/GenerationStudio.Gui/Forms/Docking/DockingForm.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using WeifenLuo.WinFormsUI.Docking;
using System.Windows.Forms;

namespace GenerationStudio.Forms.Docking
{
    public class DockingForm : DockContent
    {
        protected Control content;
        protected ToolStripContainer Container;
        protected Control oldParent;

        public DockingForm()
        {
            HideOnClose = true;
            base.DockAreas = DockAreas.DockBottom | DockAreas.DockLeft | DockAreas.DockRight | DockAreas.DockTop | DockAreas.Document | DockAreas.Float;
        }

        public virtual void SetContent(Control content, string title)
        {
            this.HideOnClose = true;
            this.content = content;
            this.oldParent = content.Parent;
            this.Controls.Clear();
            content.Parent = this;
            content.Dock = DockStyle.Fill;
            this.Text = title;
            content.Visible = true;
        }
    }
}
